agent/src/main/java/reactor/blockhound/BlockHound.java

```diff
@@ -254,6 +258,58 @@ public TypePool typePool(ClassFileLocator classFileLocator, ClassLoader classLoa
 
         private Predicate<Thread> dynamicThreadPredicate = t -> false;
 
+        /**
+         * A phase of the builder where allowed blocking methods are configured for a given class.
+         * See {@link Builder#allowBlockingCallsInside(String)}.
+         * <p>
+         * Note that methods of this class directly mutate the {@link Builder} as if one was calling
+         * {@link Builder#allowBlockingCallsInside(String, String)}. One MUST specify at least one
+         * allowed method for it to have an effect on the configuration.
+         */
+        public class BuilderAllowSpec {
```

Comment:
I don't see anything for "Disallow". Doesn't that need the same? The same builder could support both.

Reply:
it could, but I wasn't able to test the disallow feature. Either I misunderstand it or there is something wrong with the feature (which isn't tested)... so for now I prefer to focus on the allow-static-initializer case which is demonstrably supported.

Reply:
@simonbasle re "disallow": see https://github.com/reactor/BlockHound/blob/master/docs/customization.md#dis-allowing-blocking-calls-inside-methods

Reply:
I saw that and didn't find it crystal clear.
The following tests fail:

```java
public class BlockingDisallowTest {

    static {
        BlockHound.install(b -> b
                .disallowBlockingCallsInside("reactor.core.publisher.Flux", "just")
                .disallowBlockingCallsInside(NonBlockingClass.class.getName(), "example")
        );
    }

    @Test
    public void shouldDisallowFluxJust() {
        assertThatExceptionOfType(RuntimeException.class)
                .isThrownBy(() ->
                        Mono.just("one")
                            .publishOn(Schedulers.parallel())
                            .flatMapMany(i -> Flux.just("one", "two"))
                            .blockLast()
                )
                .withCauseInstanceOf(BlockingOperationError.class);

    }

    @Test
    public void shouldDisallowNonBlocking() {
        NonBlockingClass nbc = new NonBlockingClass();

        assertThatExceptionOfType(RuntimeException.class)
                .isThrownBy(() ->
                        Mono.fromCallable(nbc::example)
                            .subscribeOn(Schedulers.parallel())
                            .block()
                )
                .withCauseInstanceOf(BlockingOperationError.class);
    }

    static class NonBlockingClass {

        String example() {
            return "example";
        }
    }
}
```
What am I missing?

Reply:
You seem to be missing how BlockHound works, see https://github.com/reactor/BlockHound/blob/master/docs/how_it_works.md and especially "Blocking call decision".

tl;dr: `disallowBlockingCallsInside` disallows blocking calls inside provided method (as per the API method's name). It does not, however, mark the method as blocking, so your `Flux.just` or `NonBlockingClass.example` makes no sense.